### ISSUE TITLE
Cinnamon 3.6 → 4.0 breaks Virtualbox seamless mode: Visually smearing & missing repaints

### ISSUE THREAD

**Issue by bentolor** [linuxmint/Cinnamon#8349]
```
 * Cinnamon version: 4.0.9
 * Distribution: Mint 19.1
 * Intel Skylake, 64bit,  Intel HD Graphics 520
 * Attach /home/<username>/.xsession-errors, or /var/log/syslog
 ```

**Issue**

Virtualbox worked flawlessly for me and seamless mode is a key feature for me. Two days ago I upgraded from Linux Mint 18.3 (also kernel 4.15) to Linux Mint 19.1 (Ubuntu 18.04 bionic). Fullscreen and windowed Mode still work flawlessly, but on seamless mode is now completely broken on Cinnamon for me. This is on a multi-monitor setup.

*   Guest windows are not updated/repainted: i.e. after moving a guest window the "cutout canvas" seems to stick in position
*  taskbar etc. are not rendered/updated at all. Popup start menu only appears randomly and then is not udated any longer
*  the mini toolbar leaves a sluggish remain.
*  the screen size does not reflect the full screen estate: see the black borders on top/below.
* On entering and leaving  fullscreen mode, the window decorations are missing

**What I tried, nothing providing any remedy:**
* creating a fresh, new plain user
* Starting Cinnamon in software rendering mode
* Booting Kernel 4.15 and Kernel 4.18
* Downgrading to distributed Virtualbox 5.2 inc. guest addonds, Re-upgrading to 6.0.4 from upstream
* Switching VboxSVGA and VBoxVGA. Reinstalling GA, Switching target monitor and from/to fullscreen/seamless/windowed

**What does work:**
* Stop using Cinnamon and start a Gnome Compiz Fallback Session
* Stop using seamless mode

**Steps to reproduce**
1. Install Linux Mint 19.1 & Virtualbox (probably on Intel Hardware)
2. Start Virtualbox, Start Window 7 guest with guest additions, enter seamless mode

**Other information**
I used the same kernel/virtualbox versions under Cinnamon 3.6. So I suspect no cause there. As the problem also appears in software rendering mode but does not appear in Gnome Fallback mode.

![virtualbox-disorted](https://user-images.githubusercontent.com/588260/51910190-bba68b80-23ce-11e9-9d60-f8e360ef78fa.jpg)

**Comment by jaszhix** [comment]
Duplicate and a fix is queued for 4.2.

https://github.com/linuxmint/muffin/issues/404

**Comment by bentolor** [comment]
@jaszhix Thanks for the link: I swear: I search for tickets in advanced, but obviously in the wrong tracker…
